### PR TITLE
chore: move pardot to new component structure

### PR DESCRIPTION
**Pull request by aashishmalik**
## What are the changes introduced in this PR?

move pardot to new component structure.
New scenarios added - had no delivery tests added 11 scenarios 

## What is the related Linear task?

Resolves INT-1490

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3171/test-report.html

File: test/integrations/destinations/pardot/dataDelivery/business.ts [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
...,"salesforce_fid":"00Q6r000002LKhTPVR","crm_last_sync":"2022-01-21 18:47:37","crm_url":"https://testcompany.my.salesforce.com/00Q6r000002LKhTPVR","is_do_not_email":null,"is_do_not_call":null,"opted_out":null,"is_reviewed":1,"is_starred":null,"created_at":"2022-01-21 18:21:46","updated_at":"2022-01-21 18:48:41","campaign":{"id":42113,"name":"Test","crm_fid":"7012y000000MNOCLL4"},"assigned_to":{"user":{"id":38443703,"email":"test_rudderstack@testcompany.com","first_name":"Rudderstack","last_name":"User","job_title":null,"role":"Administrator","account":489853,"created_at":"2021-02-26 06:25:17","updated_at":"2021-02-26 06:25:17"}},"Are_you_shipping_large_fragile_or_bulky_items":false,"Calendly":false,"Country_Code":"AU","Currency":"AUD","Inventory_or_Warehouse_Management_System":false,"Lead_Status":"New","Marketing_Stage":"SAL","Record_Type_ID":"TestCompany Lead","profile":{"id":304,"name":"Default","profile_criteria":[{"id":1500,"name":"Shipping Volume","matches":"Unknown"},{"id":1502,"name":"Industry","matches":"Unknown"},{"id":1506,"name":"Job Title","matches":"Unknown"},{"id":1508,"name":"Department","matches":"Unknown"}]},"visitors":null,"visitor_activities":null,"lists":null}}',
```
Comment (koladilip) [inline test/integrations/destinations/pardot/dataDelivery/business.ts:342]: is this repeated

File: src/v0/destinations/pardot/networkHandler.js [hunk 2]
```diff
@@ -46,6 +46,20 @@ const getStatus = (code) => {
...
+  // to handle errors like service unavilable, wrong url, no response
+  if (!respAttributes) {
+    throw new NetworkError(
+      `${JSON.stringify(response)} ${stageMsg}`,
+      status,
+      {
+        [tags.TAG_NAMES.ERROR_TYPE]: getDynamicErrorType(status),
+      },
+      response,
+      getAuthErrCategory(status),
```
Comment (utsabc) [inline src/v0/destinations/pardot/networkHandler.js:59]: We are passing `http` Status but we are expecting `code` in `getAuthErrCategory` function

File: src/v0/destinations/pardot/networkHandler.js [hunk 3]
```diff
@@ -46,6 +46,20 @@ const getStatus = (code) => {
 const pardotRespHandler = (destResponse, stageMsg) => {
   const { status, response } = destResponse;
   const respAttributes = response['@attributes'];
+
+  // to handle errors like service unavilable, wrong url, no response
```
Comment (utsabc) [inline src/v0/destinations/pardot/networkHandler.js:50]: Lets add inline documentation from pardot here

File: test/integrations/destinations/pardot/dataDelivery/business.ts [hunk 4]
```diff
@@ -0,0 +1,393 @@
...
+            },
+          },
+        },
+      },
+    },
+  },
+  {
+    id: 'pardot_v0_bussiness_scenario_2',
+    name: 'pardot',
+    description: '[Proxy v0 API] :: pardot fid type upsert',
```
Comment (utsabc) [inline test/integrations/destinations/pardot/dataDelivery/business.ts:132]: The descriptions need to be more elaborative what exactly we are testing and what is the expectations?

File: test/integrations/destinations/pardot/dataDelivery/business.ts [hunk 5]
```diff
@@ -0,0 +1,393 @@
...
+} from '../../../testUtils';
+import { commonRequestParameters, retryStatTags } from './constant';
+
+export const businessV0TestScenarios = [
+  {
+    id: 'pardot_v0_bussiness_scenario_1',
+    name: 'pardot',
+    description: '[Proxy v0 API] :: pardot email upsert',
+    successCriteria: 'Proper response from destination is received',
+    scenario: 'Oauth',
```
Comment (utsabc) [inline test/integrations/destinations/pardot/dataDelivery/business.ts:15]: Doesn't seem to be oauth scenraio

File: test/integrations/destinations/pardot/dataDelivery/business.ts [hunk 6]
```diff
@@ -0,0 +1,393 @@
...
...null,"crm_contact_fid":null,"crm_owner_fid":"00G2v000004WYXaEAO","crm_account_fid":null,"salesforce_fid":null,"crm_last_sync":null,"crm_url":null,"is_do_not_email":null,"is_do_not_call":null,"opted_out":null,"is_reviewed":1,"is_starred":null,"created_at":"2022-01-21 18:21:46","updated_at":"2022-01-21 18:48:41","campaign":{"id":42113,"name":"Test","crm_fid":"7012y000000MNOCLL4"},"assigned_to":{"user":{"id":38443703,"email":"test_rudderstack@testcompany.com","first_name":"Rudderstack","last_name":"User","job_title":null,"role":"Administrator","account":489853,"created_at":"2021-02-26 06:25:17","updated_at":"2021-02-26 06:25:17"}},"Are_you_shipping_large_fragile_or_bulky_items":false,"Calendly":false,"Country_Code":"AU","Currency":"AUD","Inventory_or_Warehouse_Management_System":false,"Lead_Status":"New","Marketing_Stage":"SAL","Record_Type_ID":"TestCompany Lead","profile":{"id":304,"name":"Default","profile_criteria":[{"id":1500,"name":"Shipping Volume","matches":"Unknown"},{"id":1502,"name":"Industry","matches":"Unknown"},{"id":1506,"name":"Job Title","matches":"Unknown"},{"id":1508,"name":"Department","matches":"Unknown"}]},"visitors":null,"visitor_activities":null,"lists":null}}',
```
Comment (utsabc) [inline test/integrations/destinations/pardot/dataDelivery/business.ts:304]: This data is repeating

File: test/integrations/destinations/pardot/dataDelivery/business.ts [hunk 7]
```diff
@@ -0,0 +1,393 @@
...
+          },
+        },
+      },
+    },
+  },
+  {
+    id: 'pardot_v1_oauth_scenario_2',
+    name: 'pardot',
+    description: '[Proxy v1 API] :: Response with other retryable codes',
+    successCriteria: 'the proxy should return 500 with retryable tag',
```
Comment (utsabc) [inline test/integrations/destinations/pardot/dataDelivery/business.ts:354]: Can we add more test scnearios where internal status codes are tested example codes in attributes for which specific logic is added network handler

File: test/integrations/destinations/pardot/dataDelivery/business.ts [hunk 8]
```diff
@@ -0,0 +1,393 @@
...
+    scenario: 'Oauth',
+    feature: 'dataDelivery',
```
Comment (utsabc) [inline test/integrations/destinations/pardot/dataDelivery/business.ts:356]: Add pardot response documentation here as well on top

File: test/integrations/destinations/pardot/dataDelivery/oauth.ts [hunk 9]
```diff
@@ -0,0 +1,53 @@
+import { ProxyV1TestData } from '../../../testTypes';
+import { generateMetadata, generateProxyV1Payload } from '../../../testUtils';
+import { commonRequestParameters, retryStatTags } from './constant';
+
+export const v1OauthScenarios: ProxyV1TestData[] = [
+  {
+    id: 'pardot_v1_oauth_scenario_1',
+    name: 'pardot',
+    description:
```
Comment (utsabc) [inline test/integrations/destinations/pardot/dataDelivery/oauth.ts:9]: Lets add other scenarios like refresh token, unauthorised access scneario

**Review by utsabc — CHANGES_REQUESTED** [review]

File: test/integrations/destinations/pardot/dataDelivery/business.ts (hunk 8 above)
Comment (aashishmalik) [reply test/integrations/destinations/pardot/dataDelivery/business.ts:356]: added

File: test/integrations/destinations/pardot/dataDelivery/business.ts (hunk 4 above)
Comment (aashishmalik) [reply test/integrations/destinations/pardot/dataDelivery/business.ts:132]: here we are test email and fid upsert types call,

File: test/integrations/destinations/pardot/dataDelivery/business.ts (hunk 5 above)
Comment (aashishmalik) [reply test/integrations/destinations/pardot/dataDelivery/business.ts:15]: fixed typo

File: test/integrations/destinations/pardot/dataDelivery/business.ts (hunk 6 above)
Comment (aashishmalik) [reply test/integrations/destinations/pardot/dataDelivery/business.ts:304]: moved to constant

File: test/integrations/destinations/pardot/dataDelivery/oauth.ts (hunk 9 above)
Comment (aashishmalik) [reply test/integrations/destinations/pardot/dataDelivery/oauth.ts:9]: we only have these codes as retryable code [85, 116, 120, 121, 183, 184, 214], rest all will be aborted. have taken them in business not in oauth flow

File: src/v0/destinations/pardot/networkHandler.js (hunk 3 above)
Comment (aashishmalik) [reply src/v0/destinations/pardot/networkHandler.js:50]: in line 63, we are fetching stat from respAttributes, if it is not present will cause coding bug. Adding these scenarios to cover cases like DNS not resolved and similar

**Review by utsabc — APPROVED** [review]

**Review by mihir-4116 — APPROVED** [review]

**Pull request merged**